### PR TITLE
Fixed memory leaks

### PR DESCRIPTION
**Pull request by jesper-friis**
# Description
Fixed memory leaks. The following was done:
- Fixed leak in json_memload()
- Fixed leak in global session when running with DLITE_ATEXIT_FREE set (instead of accepting it, as suggested in PR #888)
- Added a FIXME for a leak in mapping_create_rec() in src/dlite-mappings.c. More work is required. Described in issue #894 
- Ignore external leaks in pydantic and numpy random_bit_generator
- Set DLITE_ATEXIT_FREE when running `make memcheck`. 

## Type of change
- [x] Bug fix & code cleanup
- [ ] New feature
- [ ] Documentation update
- [ ] Test update

## Checklist for the reviewer
This checklist should be used as a help for the reviewer.

- [ ] Is the change limited to one issue?
- [ ] Does this PR close the issue?
- [ ] Is the code easy to read and understand?
- [ ] Do all new feature have an accompanying new test?
- [ ] Has the documentation been updated as necessary?

### CODE REVIEW

File: src/dlite-mapping.c [hunk 1]
```diff
@@ -100,9 +100,20 @@ DLiteMapping *mapping_create_rec(const char *output_uri, Instances *inputs,
...
+    /*
+      FIXME: avoid memory leak
+      Implement dlite_mapping_plugin_free() and call it here.
+
+      Even better, implement caching in dlite_mapping_plugin_next()
+      such that it returns borrowed references to the api. Add a function
+      to clear the cache.
+    */
     if (!cheapest || cost < lowest_cost) {
+      //dlite_mapping_plugin_free(cheapest);
```
Comment (francescalb) [inline src/dlite-mapping.c:112]:
```suggestion
```

File: src/dlite-mapping.c [hunk 2]
```diff
@@ -100,9 +100,20 @@ DLiteMapping *mapping_create_rec(const char *output_uri, Instances *inputs,
...
       cheapest = api;
       lowest_cost = cost;
+    } else {
+      //dlite_mapping_plugin_free(api);
```
Comment (francescalb) [inline src/dlite-mapping.c:116]:
```suggestion
```

File: src/dlite-misc.c [hunk 3]
```diff
@@ -496,7 +497,8 @@ static Locals *get_locals(void)
 static void _handle_atexit(void) {
 
   /* No extra finalisation is needed if we already are in an atexit handler */
-  if (dlite_globals_in_atexit()) return;
+  //if (dlite_globals_in_atexit()) return;
```
Comment (francescalb) [inline src/dlite-misc.c:500]:
```suggestion
```

File: cmake/valgrind-dlite.supp [hunk 4]
```diff
@@ -220,3 +220,14 @@
    fun:expand_dynamic_string_token
    fun:_dl_map_object
 }
+
+# For Pydantic
+
+{
+   <insert_a_suppression_name_here>
```
Comment (francescalb) [inline cmake/valgrind-dlite.supp:227]: Please insert a suppression name here.

**Review by francescalb — APPROVED** [review]
Approved when comments have been adressed

File: cmake/valgrind-dlite.supp (hunk 4 above)
Comment (jesper-friis) [reply cmake/valgrind-dlite.supp]: Added comment and moved suppression to valgrind-python.supp.

**Pull request merged**